crates/router/src/connector/braintree.rs

```diff
@@ -105,11 +113,13 @@ impl ConnectorCommon for Braintree {
         res: types::Response,
         event_builder: Option<&mut ConnectorEvent>,
     ) -> CustomResult<ErrorResponse, errors::ConnectorError> {
-        let response: Result<braintree::ErrorResponse, Report<common_utils::errors::ParsingError>> =
-            res.response.parse_struct("Braintree Error Response");
+        let response: Result<
+            braintree::ErrorResponses,
+            Report<common_utils::errors::ParsingError>,
+        > = res.response.parse_struct("Braintree Error Response");
 
         match response {
-            Ok(braintree::ErrorResponse::BraintreeApiErrorResponse(response)) => {
+            Ok(braintree::ErrorResponses::BraintreeApiErrorResponse(response)) => {
                 event_builder.map(|i| i.set_error_response_body(&response));
                 router_env::logger::info!(connector_response=?response);
 
```

Comment:
We need to append multiple errors together, rather than just consuming `first`

Reply:
here multiple errors happen only when we stack multiple mutations in one request(But we are not doing that) so only first will be required